contracts/BasicPolyptychRandomizerV0.sol

```diff
@@ -0,0 +1,90 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+// Creatd By: Art Blocks Inc.
+
+pragma solidity 0.8.17;
+
+import "./interfaces/0.8.x/IRandomizerV2.sol";
+import "./interfaces/0.8.x/IGenArt721CoreContractV3.sol";
+import "./interfaces/0.8.x/IFilteredMinterHolderV1.sol";
+
+import "@openzeppelin-4.7/contracts/access/Ownable.sol";
+
+contract BasicPolyptychRandomizerV0 is IRandomizerV2, Ownable {
+    // The core contract that may interact with this randomizer contract.
+    IGenArt721CoreContractV3 public genArt721Core;
+
+    // Used to obtain the project ID from the token ID
+    uint256 constant ONE_MILLION = 1_000_000;
+
+    // Mapping of token IDs to hash seeds
+    mapping(uint256 => bytes12) public polyptychHashSeeds;
+
+    // The second/subsequent panels of polyptychs should be configured to bypass randomizing
+    mapping(uint256 => bool) projectIsPolyptych;
+
+    // The contract allowed to copy hash seeds from one token to another (PolyptychMinterV0)
+    address hashSeedSetterContract;
+
+    function assignCoreAndRenounce(address _genArt721Core) external onlyOwner {
+        renounceOwnership();
+        genArt721Core = IGenArt721CoreContractV3(_genArt721Core);
+    }
+
+    // Allows the owner of the core contract to set the minter that is allowed to copy token hashes
+    function setHashSeedSetterContract(address _contractAddress) external {
+        require(
+            genArt721Core.adminACLAllowed(
+                msg.sender,
+                address(this),
+                this.toggleProjectIsPolyptych.selector
+            ),
+            "Only Core AdminACL allowed"
+        );
+        hashSeedSetterContract = _contractAddress;
```

Comment:
Although *we* don't currently plan on indexing this value, I recommend emitting an event along the lines of `HashSeedSetterUpdated(_contractAddress)` to support any indexing that partners may want to do. We want to avoid any chance of requiring call handlers in subgraphs.